qcodes/dataset/data_set_protocol.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import annotations
+
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    List,
+    Mapping,
+    Optional,
+    Sequence,
+    Sized,
+    Tuple,
+    Union,
+)
+
+import numpy as np
+from typing_extensions import Protocol, runtime_checkable
+
+from qcodes.dataset.descriptions.dependencies import InterDependencies_
+from qcodes.dataset.descriptions.param_spec import ParamSpec, ParamSpecBase
+from qcodes.dataset.descriptions.rundescriber import RunDescriber
+from qcodes.dataset.descriptions.versioning.rundescribertypes import Shapes
+from qcodes.dataset.export_config import DataExportType
+from qcodes.dataset.linked_datasets.links import Link
+
+SPECS = List[ParamSpec]
+# Transition period type: SpecsOrInterDeps. We will allow both as input to
+# the DataSet constructor for a while, then deprecate SPECS and finally remove
+# the ParamSpec class
+SpecsOrInterDeps = Union[SPECS, InterDependencies_]
+
+
+if TYPE_CHECKING:
+    from qcodes.station import Station
+
+    from .data_set_cache import DataSetCache
+
+
+@runtime_checkable
+class DataSetProtocol(Protocol, Sized):
+    def prepare(
+        self,
+        *,
+        station: Optional[Station],
```

Comment:
is it necessary for the dataset to know about the station? can't we pass the snapshot (as a str) directly to ``prepare``? this would be consistent with the fact that ``add_snapshot`` takes a string, and ``snapshot_raw`` is the only way to get that snapshot back (no ``snapshot`` property is defined)

Reply:
This is how it was done so far but you are right we should revisit that and see if it still makes sense

Reply:
We now pass the snapshot. Please have a look and see if that is better